dask_expr/io/tests/test_io.py

```diff
@@ -207,6 +207,16 @@ def test_parquet_complex_filters(tmpdir):
     assert_eq(got.optimize(), expect)
 
 
+def test_parquet_len(tmpdir):
+    df = read_parquet(_make_file(tmpdir))
+    pdf = df.compute()
+
+    assert len(df[df.a > 5]) == len(pdf[pdf.a > 5])
+
+    s = (df["b"] + 1).astype("Int32")
+    assert len(s) == len(pdf)
```

Comment:
Do you want to verify that we can do this without any actual execution somehow?

Maybe we can call something like the following

```python
expr = Len(s)
assert isinstance(s.optimize(), Literal)
```

Reply:
good call